onecall.go

```diff
@@ -0,0 +1,176 @@
+	Pressure    int       `json:"pressure"`
+	Humidity    int       `json:"humidity"`
+	DewPoint    float64   `json:"dew_point"`
+	Clouds      int       `json:"clouds"`
+	UVI         float64   `json:"uvi"`
+	Visibility  int       `json:"visibility"`
+	WindSpeed   float64   `json:"wind_speed"`
+	WindGust    float64   `json:"wind_gust,omitempty"`
+	WindDeg     float64   `json:"wind_deg"`
+	Rain        Rain      `json:"rain,omitempty"`
+	Snow        Snow      `json:"snow,omitempty"`
+	Weather     []Weather `json:"weather"`
+}
+
+type OneCallMinutelyData struct {
+	Dt            int     `json:"dt"`
+	Precipitation float64 `json:"precipitation"`
+}
+
+type OneCallHourlyData struct {
+	Dt          int       `json:"dt"`
+	Temperature float64   `json:"temp"`
+	FeelsLike   float64   `json:"feels_like"`
+	Pressure    int       `json:"pressure"`
+	Humidity    int       `json:"humidity"`
+	DewPoint    float64   `json:"dew_point"`
+	Uvi         float64   `json:"uvi"`
+	Clouds      int       `json:"clouds"`
+	Visibility  int       `json:"visibility"`
+	WindSpeed   float64   `json:"wind_speed"`
+	WindGust    float64   `json:"wind_gust,omitempty"`
+	WindDeg     float64   `json:"wind_deg"`
+	Pop         float64   `json:"pop"`
+	Rain        Rain      `json:"rain,omitempty"`
+	Snow        Snow      `json:"snow,omitempty"`
+	Weather     []Weather `json:"weather"`
+}
+
+type OneCallDailyData struct {
+	Dt        int         `json:"dt"`
+	Sunrise   int         `json:"sunrise"`
+	Sunset    int         `json:"sunset"`
+	Moonrise  int         `json:"moonrise"`
+	Moonset   int         `json:"moonset"`
+	MoonPhase float64     `json:"moon_phase"`
+	Temp      Temperature `json:"temp"`
+	FeelsLike struct {
+		Day   float64 `json:"day"`
+		Night float64 `json:"night"`
+		Eve   float64 `json:"eve"`
+		Morn  float64 `json:"morn"`
+	} `json:"feels_like"`
+	Pressure  int       `json:"pressure"`
+	Humidity  int       `json:"humidity"`
+	DewPoint  float64   `json:"dew_point"`
+	WindSpeed float64   `json:"wind_speed"`
+	WindGust  float64   `json:"wind_gust,omitempty"`
+	WindDeg   float64   `json:"wind_deg"`
+	Clouds    int       `json:"clouds"`
+	Uvi       float64   `json:"uvi"`
+	Pop       float64   `json:"pop"`
+	Rain      float64   `json:"rain,omitempty"`
+	Snow      float64   `json:"snow,omitempty"`
+	Weather   []Weather `json:"weather"`
+}
+
+type OneCallAlertData struct {
+	SenderName  string   `json:"sender_name"`
+	Event       string   `json:"event"`
+	Start       int      `json:"start"`
+	End         int      `json:"end"`
+	Description string   `json:"description"`
+	Tags        []string `json:"tags"`
+}
+
+// NewCurrent returns a new OneCallData pointer with the supplied parameters
+func NewOneCall(unit, lang, key string, excludes []Exclude, options ...Option) (*OneCallData, error) {
+	unitChoice := strings.ToUpper(unit)
+	langChoice := strings.ToUpper(lang)
+
+	c := &OneCallData{
+		Settings: NewSettings(),
+	}
+
+	if ValidDataUnit(unitChoice) {
```

Comment:
Can you reverse the logic here? Reason is that we can get rid of the 'else' block and do the assignment of `c.Unit` and keep processing.

Reply:
And the one below.

Reply:
I eliminated the else [here](https://github.com/briandowns/openweathermap/pull/101/files#diff-827c790c48669d39c34193629e0752dab0bec2630a7b7a9d66fbd2d7ce7388cfR132)